lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1677,6 +1678,25 @@ optimizeBridgedSwiftToObjCCast(SILInstruction *Inst,
   SILType SubstFnTy = SILFnTy.substGenericArgs(M, SubMap);
   SILFunctionConventions substConv(SubstFnTy.castTo<SILFunctionType>(), M);
 
+  if (Dest) {
```

Comment:
What if Dest is a null (which is the case for the non-addr variation of cast instructions)? Can we hit the bug also in this cases?
In other words: shouldn't you do this check also with a null Dest value and use TargetType instead?